### PR TITLE
fix: add isNested prop to ThemeProvider

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

Add `isNested` prop to `ThemeProvider` so that only the one at top will be able to set attributes on root element versus relying on the cleanup function in `useEffect`, which will not work as expected if there are 2 or more nested providers on the same page.

#### Issue #, if available

A good example I can see is on the [Authenticator page](https://dev.ui.docs.amplify.aws/react/components/authenticator) where 2 ThemeProvider are in use. If you go to that page and then move over to other pages like [Installation](https://dev.ui.docs.amplify.aws/react/getting-started/installation), you will see `data-amplify-theme` attribute on root element is set to a wrong value `Auth Example Theme` instead of the expected original value `amplify-docs`.

#### Description of how you validated changes

Tested it out locally

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/ThemeProvider/index.tsx [hunk 1]
```diff
@@ -29,31 +37,15 @@ export function AmplifyProvider({
   // the same data attributes are on a div down the DOM tree, the CSS variables
   // will apply to both.
   React.useEffect(() => {
-    if (document && document.documentElement) {
-      // Keep original data attributes to reset on unmount
```
Comment (joebuono) [inline packages/react/src/components/ThemeProvider/index.tsx:33]: Do we no longer need to keep original data attributes to reset on unmount?

**Review by joebuono — APPROVED** [review]
Just one question. Thanks for making this change!

**Comment by zchenwei** [comment]
> Let's pause on this, @calebpollman has some concerns

What is the concern?

**Comment by zchenwei** [comment]
Close it in favor of #2102

**Pull request closed without merging**